diff --git a/src/dsp/cpu.c b/src/dsp/cpu.c
index ef04a75..aab68a9 100644
--- a/src/dsp/cpu.c
+++ b/src/dsp/cpu.c
@@ -13,10 +13,6 @@
 
 #include "./dsp.h"
 
-#if defined(__ANDROID__)
-#include <cpu-features.h>
-#endif
-
 //------------------------------------------------------------------------------
 // SSE2 detection.
 //
@@ -104,16 +100,7 @@ static int x86CPUInfo(CPUFeature feature) {
 }
 VP8CPUInfo VP8GetCPUInfo = x86CPUInfo;
 #elif defined(WEBP_ANDROID_NEON)  // NB: needs to be before generic NEON test.
-static int AndroidCPUInfo(CPUFeature feature) {
-  const AndroidCpuFamily cpu_family = android_getCpuFamily();
-  const uint64_t cpu_features = android_getCpuFeatures();
-  if (feature == kNEON) {
-    return (cpu_family == ANDROID_CPU_FAMILY_ARM &&
-            0 != (cpu_features & ANDROID_CPU_ARM_FEATURE_NEON));
-  }
-  return 0;
-}
-VP8CPUInfo VP8GetCPUInfo = AndroidCPUInfo;
+VP8CPUInfo VP8GetCPUInfo = NULL;
 #elif defined(WEBP_USE_NEON)
 // define a dummy function to enable turning off NEON at runtime by setting
 // VP8DecGetCPUInfo = NULL
